src/parser/expression.js

```diff
@@ -1118,8 +1118,11 @@ export default class ExpressionParser extends LValParser {
 
       if (this.hasPlugin("objectRestSpread") && this.match(tt.ellipsis)) {
         prop = this.parseSpread(isPattern ? { start: 0 } : undefined);
-        prop.type = isPattern ? "RestElement" : "SpreadElement";
-        if (isPattern) this.toAssignable(prop.argument, true, "object pattern");
+        if (isPattern) {
+          this.checkToRestConversion(prop);
```

Comment:
this probably could be forbidden somehow [here](https://github.com/babel/babylon/blob/88cf1bc140d19da19e9496b6293103d992bd9e85/src/parser/lval.js#L58) and [here](https://github.com/babel/babylon/blob/88cf1bc140d19da19e9496b6293103d992bd9e85/src/parser/lval.js#L53), but I would need to have knowledge there is im in declaration context **and** im checking rest's argument and Im not entirely sure whats the best practice to share such information